src/mainwindow.cpp

```diff
@@ -3118,41 +3118,51 @@ void MainWindow::displayModInformation(int row, ModInfoTabIDs tabID)
 
 void MainWindow::ignoreMissingData_clicked()
 {
-  QItemSelectionModel *selection = ui->modList->selectionModel();
-  if (selection->hasSelection() && selection->selectedRows().count() > 1) {
-    for (QModelIndex idx : selection->selectedRows()) {
+  const auto rows = ui->modList->selectionModel()->selectedRows();
+
+  if (rows.count() > 1) {
+    std::vector<ModInfo::Ptr> changed;
+
+    for (QModelIndex idx : rows) {
       int row_idx = idx.data(Qt::UserRole + 1).toInt();
       ModInfo::Ptr info = ModInfo::getByIndex(row_idx);
       info->markValidated(true);
-      connect(this, SIGNAL(modListDataChanged(QModelIndex, QModelIndex)), m_OrganizerCore.modList(), SIGNAL(dataChanged(QModelIndex, QModelIndex)));
+      changed.push_back(info);
+    }
 
-      emit modListDataChanged(m_OrganizerCore.modList()->index(row_idx, 0), m_OrganizerCore.modList()->index(row_idx, m_OrganizerCore.modList()->columnCount() - 1));
+    for (auto&& m : changed) {
+      int row_idx = ModInfo::getIndex(m->internalName());
+      m_OrganizerCore.modList()->notifyChange(row_idx);
     }
   } else {
     ModInfo::Ptr info = ModInfo::getByIndex(m_ContextRow);
     info->markValidated(true);
-    connect(this, SIGNAL(modListDataChanged(QModelIndex, QModelIndex)), m_OrganizerCore.modList(), SIGNAL(dataChanged(QModelIndex, QModelIndex)));
-
-    emit modListDataChanged(m_OrganizerCore.modList()->index(m_ContextRow, 0), m_OrganizerCore.modList()->index(m_ContextRow, m_OrganizerCore.modList()->columnCount() - 1));
+    m_OrganizerCore.modList()->notifyChange(m_ContextRow);
   }
 }
 
 void MainWindow::markConverted_clicked()
 {
-  QItemSelectionModel *selection = ui->modList->selectionModel();
-  if (selection->hasSelection() && selection->selectedRows().count() > 1) {
-    for (QModelIndex idx : selection->selectedRows()) {
+  const auto rows = ui->modList->selectionModel()->selectedRows();
+
+  if (rows.count() > 1) {
+    std::vector<ModInfo::Ptr> changed;
+
+    for (QModelIndex idx : rows) {
       int row_idx = idx.data(Qt::UserRole + 1).toInt();
       ModInfo::Ptr info = ModInfo::getByIndex(row_idx);
       info->markConverted(true);
-      connect(this, SIGNAL(modListDataChanged(QModelIndex, QModelIndex)), m_OrganizerCore.modList(), SIGNAL(dataChanged(QModelIndex, QModelIndex)));
-      emit modListDataChanged(m_OrganizerCore.modList()->index(row_idx, 0), m_OrganizerCore.modList()->index(row_idx, m_OrganizerCore.modList()->columnCount() - 1));
+      changed.push_back(info);
+    }
+
+    for (auto&& m : changed) {
+      int row_idx = ModInfo::getIndex(m->internalName());
+      m_OrganizerCore.modList()->notifyChange(row_idx);
```

Comment:
If selection is sequential wouldn't it be better to use the notifyChange(startRow, endRow) ?  This is the kind of thing that causes stuff to refresh multiple times in a row

Reply:
Maybe. When the list is filtered, the rows can disappeared, and I'm not sure when or where, so I don't know whether a range would work. I'm not too keen on trying it.

Reply:
Another thing is that by using notifyChange you are forcing a recalculation of the conflicts of the mods in question, while before it was emitting dataChanged() with the weird connection which (I don't think) forced the conflicts refresh.

Reply:
All `notifyChange()` does is fire `dataChanged()`, there's no change afaict.

Reply:
You are right I read notifychange wrong, it does the conflict checks again only if it's passed -1.